csharp/lib/src/lib.rs

```diff
@@ -93,6 +93,257 @@ pub extern "C" fn close_client(client_ptr: *const c_void) {
+        RequestType::RPush => Some(cmd("RPUSH")),
+        RequestType::RPop => Some(cmd("RPOP")),
+        RequestType::LLen => Some(cmd("LLEN")),
+        RequestType::LRem => Some(cmd("LREM")),
+        RequestType::LRange => Some(cmd("LRANGE")),
+        RequestType::LTrim => Some(cmd("LTRIM")),
+        RequestType::SAdd => Some(cmd("SADD")),
+        RequestType::SRem => Some(cmd("SREM")),
+        RequestType::SMembers => Some(cmd("SMEMBERS")),
+        RequestType::SCard => Some(cmd("SCARD")),
+        RequestType::PExpireAt => Some(cmd("PEXPIREAT")),
+        RequestType::PExpire => Some(cmd("PEXPIRE")),
+        RequestType::ExpireAt => Some(cmd("EXPIREAT")),
+        RequestType::Exists => Some(cmd("EXISTS")),
+        RequestType::Unlink => Some(cmd("UNLINK")),
+        RequestType::TTL => Some(cmd("TTL")),
+        RequestType::Zadd => Some(cmd("ZADD")),
+        RequestType::Zrem => Some(cmd("ZREM")),
+        RequestType::Zrange => Some(cmd("ZRANGE")),
+        RequestType::Zcard => Some(cmd("ZCARD")),
+        RequestType::Zcount => Some(cmd("ZCOUNT")),
+        RequestType::ZIncrBy => Some(cmd("ZINCRBY")),
+        RequestType::ZScore => Some(cmd("ZSCORE")),
+        RequestType::Type => Some(cmd("TYPE")),
+        RequestType::HLen => Some(cmd("HLEN")),
+        RequestType::Echo => Some(cmd("ECHO")),
+        RequestType::ZPopMin => Some(cmd("ZPOPMIN")),
+        RequestType::Strlen => Some(cmd("STRLEN")),
+        RequestType::Lindex => Some(cmd("LINDEX")),
+        RequestType::ZPopMax => Some(cmd("ZPOPMAX")),
+        RequestType::XAck => Some(cmd("XACK")),
+        RequestType::XAdd => Some(cmd("XADD")),
+        RequestType::XReadGroup => Some(cmd("XREADGROUP")),
+        RequestType::XRead => Some(cmd("XREAD")),
+        RequestType::XGroupCreate => Some(get_two_word_command("XGROUP", "CREATE")),
+        RequestType::XGroupDestroy => Some(get_two_word_command("XGROUP", "DESTROY")),
+        RequestType::XTrim => Some(cmd("XTRIM")),
+    }
+}
+
+// copied from glide_core::socket_listener::get_two_word_command
+fn get_two_word_command(first: &str, second: &str) -> Cmd {
+    let mut cmd = cmd(first);
+    cmd.arg(second);
+    cmd
+}
+
+use std::slice::from_raw_parts;
+use std::str::Utf8Error;
+
+pub unsafe fn convert_double_pointer_to_vec(
+    data: *const *const c_char,
+    len: usize,
+) -> Result<Vec<String>, Utf8Error> {
+    from_raw_parts(data, len)
+        .iter()
+        .map(|arg| CStr::from_ptr(*arg).to_str().map(ToString::to_string))
+        .collect()
+}
+
+#[no_mangle]
+pub extern "C" fn command(
+    client_ptr: *const c_void,
+    callback_index: usize,
+    command_type: RequestType,
+    arg_count: usize,
+    args: *const *const c_char
+) {
+    let client = unsafe { Box::leak(Box::from_raw(client_ptr as *mut Client)) };
+    // The safety of this needs to be ensured by the calling code. Cannot dispose of the pointer before all operations have completed.
```

Comment:
If the safety of `command` needs to ensured by the calling code, then `command` should be marked `unsafe fn`